File: AoC2021.CSharp/Day0.Alternative.cs
using System.Text;

using AoC2021.Core;

using Xunit;

namespace AoC2021.CSharp;

[Solver("day0", "stringbuilder")]
public class Day0Alternative : Solver {
    public Day0Alternative(string? input = null) : base(input) { }

    public override string SolvePartOne() {
        var sb = new StringBuilder(32);
        sb.Append("Day 0 C# Alternative: ");
        sb.Append(Input);
        return sb.ToString();
    }

    [Fact]
    public void AlternativeSolvesExample() {
        const string exampleInput = @"foo";

        var solver = new Day0Alternative(exampleInput);
        var actual = solver.SolvePartOne();

        Assert.Equal("Day 0 C# Alternative: foo", actual);
    }
}

File: AoC2021.FSharp/Day0.Alternative.fs
namespace AoC2021.FSharp

open AoC2021.Core
open Xunit

module Day0Alternative =
    let Solve = sprintf "Day 0 F#: %s"

    [<Fact>]
    let ``Solves Example`` () =
        let exampleInput = @"foo"

        let actual = Solve exampleInput
        Assert.Equal("Day 0 F#: foo", actual)

[<Solver("day0", "sprintf")>]
type Day0Alternative(input: string) =
    interface ISolver with
        member this.SolvePartOne() = Day0.Solve input
        member this.SolvePartTwo() = failwith "NotImplemented"
